test/ZZStRSR.test.ts

```diff
@@ -1706,6 +1707,43 @@ describe(`StRSRP${IMPLEMENTATION} contract`, () => {
       expect(await stRSR.balanceOf(addr1.address)).to.equal(amount.sub(one))
       expect(await stRSR.totalSupply()).to.equal(amount.sub(one))
     })
+
+    it('Should not allow stakeRate manipulation', async () => {
+      // send RSR to stRSR (attempt to manipulate stake rate)
+      await rsr.connect(addr1).transfer(stRSR.address, fp('200'))
+      const attackerBalBefore = await rsr.balanceOf(addr1.address)
+      const stakerBalBefore = await rsr.balanceOf(addr2.address)
+
+      // stake 1 wei
+      await rsr.connect(addr1).approve(stRSR.address, bn(1))
+      await stRSR.connect(addr1).stake(bn(1))
+
+      // wait, then payout rewards
+      await advanceTime(12) // 12 second reward period
+      await stRSR.payoutRewards()
```

Comment:
I think before and after this line we want to read out `exchangeRate()` and assert it does not change

Reply:
Also we've had trouble with `advanceTime` when dealing with small numbers. Might want to use `await setNextBlockTimestamp((await getLatestBlockTimestamp()) + 12)`